src/peft/tuners/lora.py

```diff
@@ -116,9 +116,31 @@ class LoraConfig(PeftConfig):
             "help": "The layer pattern name, used only if `layers_to_transform` is different to None and if the layer pattern is not in the common layers pattern."
         },
     )
+    rank_pattern: Optional[dict] = field(
+        default=None,
```

Comment:
Instead of `default=None`, you could use `default_factory=dict`, then you don't need the extra `__post_init__`:

https://docs.python.org/3/library/dataclasses.html#dataclasses.field

Reply:
Done